src/java.desktop/macosx/native/libosxapp/JNIUtilities.h

```diff
@@ -29,11 +29,18 @@
 #include "jni.h"
 #include "jni_util.h"
 
+#import <Cocoa/Cocoa.h>
+
 /********        LOGGING SUPPORT    *********/
 
 #define LOG_NULL(dst_var, name) \
    if (dst_var == NULL) { \
        NSLog(@"Bad JNI lookup %s\n", name); \
+       NSLog(@"%@",[NSThread callStackSymbols]); \
+       if ([NSThread isMainThread] == NO) { \
+           JNU_ThrowInternalError(env, "Bad JNI Lookup"); \
```

Comment:
It will be possible that(most of the time) the JNU_ThrowInternalError will be called while we already have a raised java exception.

Reply:
I don't think "most" of the time is very likely. JNI lookup failures don't cause exceptions to be thrown.
Is that what you are tihinking ?

And separately, with either the current code of clearing exceptions or the change here to  the practice of throwing NSException on seeing a Java Exception I think it very unlikely.

Reply:
Then how we get NoSuchMethodError here? https://bugs.openjdk.java.net/browse/JDK-8259232?
https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/functions.html#GetMethodID
Calling Instance Methods: GetMethodID
RETURNS:
    Returns a method ID, or NULL if the specified method cannot be found.
THROWS:
    NoSuchMethodError: if the specified method cannot be found.
    ExceptionInInitializerError: if the class initializer fails due to an exception.
    OutOfMemoryError: if the system runs out of memory.

Same for fields:
https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/functions.html#GetFieldID

Reply:
Ok. I will only create my own if none is pending.